docs/advanced-topics/best-practices/best-practices-dockstore.rst

```diff
@@ -0,0 +1,107 @@
+Dockstore Best Practices
+========================
+
+This document is intended for tool and workflow developers and the users who register them on Dockstore. Following these best practices will help make
+your repositories more organized, findable, and usable for others on Dockstore.
+
+
+Structuring and Organizing Your Git Repositories
+----------------------------------------------------------------
+Unless you're writing your tool/workflow directly on Dockstore, your entry will be hosted on a Git repository (we currently support integration with GitHub, BitBucket, and Gitlab).
+Whichever repository service you use, we recommend that your tool and workflow repositories are put under a non-personal organization, if possible.
+This allows for better organization and collaboration, and also provides a fallback for others if you become inactive on the git repository site.
+
+We also highly recommend that you do NOT host different tools and workflows on a single repo by putting them on different branches.
+Our reasoning is that branches are supposed to be used when working on a new feature or bug fix. That means they should eventually be merged with the repository's main branch.
+Another reason why we are against this practice is because it can lead to a lot of clutter on the Dockstore entries themselves.
+Let's say you have three workflows in a single repo on separate branches and there are many tags/releases related to each one.
+Depending on how you register the workflows on Dockstore, each entry could have the other two branches and every single tag/release
+on its Versions tab. All of this clutter can be overwhelming or confusing for users to see.
+
+Likewise, we don't recommend using even a single branch to hosts multiple entries. A repository should essentially represent one project, but that practice can be a grey area.
```

Comment:
Grammatically,  "hosts" should be "host", but I would say "contain"

Regarding this and the previous paragraph, seems like we are taking a hard stance a common pattern. I agree with the previous paragraph more; this one I'm not so sure. The reason people do it this way it to share code.

Reply:
I wrote this section based off memory of conversations (not exactly reliable)... I thought we almost never liked when users had multiple workflows in a repo. If it's just not recommended when they use separate branches, then I can remove this paragraph.

Reply:
It's caused us some issues server-side, but it seems to be a fairly common pattern. Whether we want to recommend for or against it, I'm not sure.

Reply:
I'd say soft-recommend against. 
I would say that 
a) putting workflows in different repos gives shorter names since you don't need an extra name to distinguish
b) you don't get weird extra releases when you release/tag one workflow but don't really mean to refresh the other one

But it seems like people like multiple workflows in one repo to share code, so :shrug: